### PR TITLE
Add og:meta tags to stats page

### PR DESCRIPTION
**Pull request by l1n**
For #1434

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -174,6 +174,21 @@ def delete_history
...
+    post_description = generate_short(@post.description)
+    post_description += ' ('
+    post_description += helpers.author_links(@post, linked: false)
+    post_description += ')'
+    post_description.strip!
+
+    @meta_og = {
+      title: @post.subject + ' · ' + post_location,
+      description: post_description,
+      url: @meta_canonical
```
Comment (Throne3d) [inline app/controllers/posts_controller.rb:190]: I think this variable isn't set to anything in posts#stats, is it?

**Review by Throne3d — COMMENTED** [review]
I was looking at the code and wondering if we could join it with the `og_data_for_post` function in `writable_controller`, but I think it's not worth the effort (too many differences that are important). Looks good to me, but marking comment because I'm unsure of the url field (other comment)

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (teceler) [reply app/controllers/posts_controller.rb:190]: idk, how does that get set?

Comment (Throne3d) [reply app/controllers/posts_controller.rb:190]: I could only see it declared in `def show_post(cur_page=nil)` on writable_controller – which I think we don't call for the stats page

Comment (l1n) [reply app/controllers/posts_controller.rb:190]: Based on the test at `spec/controllers/post_controller_spec.rb`, line 844 it should be set.

Comment (Throne3d) [reply app/controllers/posts_controller.rb:191]: That line's not showing anything relevant for me: https://github.com/Marri/glowfic/blob/aa2ceed9e72deca3788f0b6e892a1b9dd7d40de5/spec/controllers/posts_controller_spec.rb#L844 (maybe it was different at a previous commit?)

Comment (l1n) [reply app/controllers/posts_controller.rb:191]: Whoops, misread it as the :show test - updating with a test and setting the url.

File: spec/controllers/posts_controller_spec.rb [hunk 2]
```diff
@@ -1433,6 +1433,18 @@ def restore(reply)
...
     end
 
+    it "calculates OpenGraph meta" do
+      user = create(:user, username: 'example user')
+      board = create(:board, name: 'board')
+      post = create(:post, subject: 'title', user: user, board: board)
+      get :stats, params: { id: post.id }
+
+      meta_og = assigns(:meta_og)
+      expect(meta_og[:url]).to eq(post_url(post))
```
Comment (Throne3d) [inline spec/controllers/posts_controller_spec.rb:1443]:
```suggestion
      expect(meta_og[:url]).to eq(stats_post_url(post))
```

**Review by Throne3d — COMMENTED** [review]
Looks good after this final change, though!

**Review by Throne3d — APPROVED** [review]

**Comment by Throne3d** [comment]
I'm not 100% sure this works right with fresh_when (does it need to be called first? last?), but I'll see if I can do some local testing on it quickly before approving!

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -176,6 +176,22 @@ def delete_history
...
+    post_description = generate_short(@post.description)
+    post_description += ' ('
+    post_description += helpers.author_links(@post, linked: false)
+    post_description += ')'
+    post_description.strip!
+
+    @meta_og = {
+      title: @post.subject + ' · ' + post_location,
+      description: post_description,
+      url: stats_post_url(@post)
```
Comment (Throne3d) [inline app/controllers/posts_controller.rb:192]:
```suggestion
      url: stats_post_url(@post),
```

**Comment by Throne3d** [comment]
The fresh_when thing seems to work both before and after, which I guess makes sense anyway since meta_og is only used in page render afterwards.

Seems fine!

**Pull request merged**